gateways/block_api_gateway.py

```diff
@@ -0,0 +1,58 @@
+from typing import Optional
+
+from elasticsearch import Elasticsearch
+
+from common.configuration import ELASTIC_TX_INDEX
+from gateways.clients.elastic_search_client import ElasticSearchClient
+from utils.elastic_search.elastic_search_utils import ElasticSearchUtils
+
+BLOCK_TX_VERSION = 0
+
+
+class BlockApiGateway:
+    """ Gateway to interact with ElasticSearch Cluster
+        To get information about Blocks
+    """
+
+    def __init__(
+        self,
+        elastic_search_client: Optional[Elasticsearch] = None
+    ) -> None:
+        self.elastic_search_client = ElasticSearchClient(elastic_index=ELASTIC_TX_INDEX,
+                                                         client=elastic_search_client
+                                                         )
+
+    def get_block_with_biggest_height(self) -> dict:
+        """ Get the block with biggest height available on the ElasticSearch
+        """
+
+        body = {
+            'index': ELASTIC_TX_INDEX,
+            'query': {
+                'bool': {
+                    'must': [
+                        {
+                            'match': {
+                                'version': BLOCK_TX_VERSION,
+                            }
+                        },
+                        {
+                            'match': {
+                                'voided': False
+                            }
+                        }
+                    ],
+                },
+            },
+            'sort': [
+                {
+                    'height': {
+                        'order': 'desc'
+                    }
+                }
+            ],
+            'size': 1
+        }
+
+        elastic_search_utils = ElasticSearchUtils(elastic_index=ELASTIC_TX_INDEX)
+        return elastic_search_utils.treat_response(self.elastic_search_client.run(body))
```

Comment:
I think we are not treating errors that could happen in the elasticsearch client, right?

Like the ones in https://elasticsearch-py.readthedocs.io/en/v8.2.2/exceptions.html

Shouldn't we treat them to prevent the service from returning 500?

Reply:
Done in last review. I've caught some of the exceptions from ElasticSearch and gave meaningful message to the client.